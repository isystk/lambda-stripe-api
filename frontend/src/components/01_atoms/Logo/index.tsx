import React, { FC } from 'react'
import { ContainerProps, WithChildren } from 'types'
import { connect } from '@/components/hoc'
import * as styles from './styles'
import { APP_NAME } from '@/constants'
import { Url } from '@/constants/url'

/** LogoProps Props */
export type LogoProps = WithChildren & {
  name?: string
  link?: string
}
/** Presenter Props */
export type PresenterProps = LogoProps

/** Presenter Component */
const LogoPresenter: FC<PresenterProps> = ({
  name = APP_NAME,
  link = Url.Top,
}) => (
  <>
    <a href={link} className={styles.logo}>
      <img src="/images/logo.png" alt={name} className="w-20 md:w-40"></img>
    </a>
  </>
)

/** Container Component */
const LogoContainer: React.FC<ContainerProps<LogoProps, PresenterProps>> = ({
  presenter,
  children,
  ...props
}) => {
  return presenter({
    children,
    ...props,
  })
}

export default connect<LogoProps, PresenterProps>(
  'Logo',
  LogoPresenter,
  LogoContainer
)
