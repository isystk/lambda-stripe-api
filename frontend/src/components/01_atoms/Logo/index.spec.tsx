import React from 'react'
import { render } from '@testing-library/react'
import Logo, { LogoProps } from './index'

import { useRouter } from 'next/router'

jest.mock('next/router', () => ({
  useRouter: jest.fn(),
}))


describe('LandingPageTemplate', () => {
  it('Match Snapshot', () => {
    useRouter.mockImplementationOnce(() => ({
      route: '/',
      pathname: '/',
      query: {},
      asPath: '/',
    }))
    const props: LogoProps = {
      name: 'sample',
      link: '#',
    }
    const { asFragment } = render(<Logo {...props} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
