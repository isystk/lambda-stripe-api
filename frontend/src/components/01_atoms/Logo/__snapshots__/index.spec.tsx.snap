// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo Match Snapshot 1`] = `
<a
  className="css-0"
  href="/"
>
  <img
    alt="sample"
    className="w-20 md:w-40"
    src="/images/logo.png"
  />
</a>
`;
